### ISSUE TITLE
Create consent metadata macros for RTC vendors

### ISSUE THREAD

**Issue by micajuine-ho** [ampproject/amphtml#30006]
Original discussion: https://github.com/ampproject/amphtml/issues/26229#issuecomment-676544721

With the introduction of [consent metadata](https://github.com/ampproject/amphtml/blob/master/extensions/amp-consent/integrating-consent.md#consentmetadata), integrated ad vendors can access this information via our [internal APIs](https://github.com/ampproject/amphtml/blob/master/src/consent.js).

When we originally created the metadata field, we agreed that we would not add macro support until it was requested; so we are going to move forward with that. 

Proposal: 
- Create and register CONSENT_METADATA macro in both amp-analytics and amp-a4a in the RTC
   `CONSENT_METADATA(key)` => returns the value from the corresponding _key_ in the [ConsentMetadata](https://github.com/ampproject/amphtml/blob/master/extensions/amp-consent/integrating-consent.md#consentmetadata) object. Will error if key is undefined. 

amp-analytics Macro Design
- Register CONSENT_METADATA(key) macro as a part of the default macros in amp-analytics 
- Use internal API to get consent metadata
- Return the associated value from the key
- Like CONSENT_STATE, return null if no consent is returned from our internal API

amp-a4a RTC Macro Design
- Expand amp-a4a to use the whole ConsentMetadata object (instead of just `gdprApplies`) when we try to get the ad url
- Register CONSENT_METADATA macro to be accessible to all implementations of amp-a4a
- Like CONSENT_STATE and CONSENT_STRING, return null if there is no CONSENT_METADATA object passed from our internal APIs
- If key is not defined, return a dev error that key must be defined, and then return null

/cc @zhouyx

**Comment by bretg** [comment]
So to make sure we understand the proposal here, we could update our RTC URLs to include these macros like this:

```
  prebidappnexus: {
    url:
      'https://prebid.adnxs.com/pbs/v1/openrtb2/amp?...&consent_type=CONSENT_METADATA(consentStringType)&consent_string=CONSENT_STRING&gdpr_applies=CONSENT_METADATA(gdprApplies)&attl_consent=CONSENT_METADATA(additionalConsent)',
   ...
  },
```

correct?

**Comment by micajuine-ho** [comment]
Yes that's correct

**Comment by micajuine-ho** [comment]
Closed via #30152
